src/software/jetson_nano/services/motor.cpp

```diff
@@ -624,6 +629,38 @@ void MotorService::spiTransfer(int fd, uint8_t const* tx, uint8_t const* rx, uns
 }
 
 
+void MotorService::readThenWriteSpiTransfer(int fd, const uint8_t* read_tx,
+                                            const uint8_t* write_tx,
+                                            const uint8_t* read_rx, uint32_t spi_speed)
+{
+    uint8_t write_rx[5] = {0};
+
+    struct spi_ioc_transfer tr[2];
+    memset(tr, 0, sizeof(tr));
+
+    tr[0].tx_buf        = (unsigned long)read_tx;
```

Comment:
just a question: is there a link to docs or api for why it's called `tr`? I didn't see anything in `motor.cpp/h` other than refs to the chip name, or is it somewhere else I didn't look?

Reply:
tr for `transaction` or `transfer` I believe